Gems/Atom/RHI/Vulkan/Code/Source/RHI/StreamingImagePool.cpp

```diff
@@ -76,11 +76,18 @@ namespace AZ
                 return RHI::ResultCode::OutOfMemory;
             }
 
+            auto memoryView = m_memoryAllocator.Allocate(memoryRequirements.size, memoryRequirements.alignment);
+            if (!memoryView.IsValid())
+            {
+                memoryUsage.m_reservedInBytes -= memoryRequirements.size;
```

Comment:
Can the `memoryRequirments.size` ever be larger than `memoryUsage.m_reservedInBytes`?
Do we need to protect against < 0 underflow?

Reply:
> Can the memoryRequirments.size ever be larger than memoryUsage.m_reservedInBytes?
Do we need to protect against < 0 underflow?

It can't. 
The `memoryRequitements.size` is added to the `memoryUsage.m_reservedInBytes` value immediately above this in the call to TryReserveMemory, and would exit this function before reaching here if `memoryRequitements.size` was larger than the available bytes in the memory allocator.